apps/autoscheduler/test.cpp

```diff
@@ -1,50 +1,13 @@
 #include "Halide.h"
 
-#if defined(_MSC_VER) && !defined(NOMINMAX)
-#define NOMINMAX
-#endif
-#ifdef _WIN32
-#include <windows.h>
-#else
-#include <dlfcn.h>
-#endif
-
 using namespace Halide;
 
 int main(int argc, char **argv) {
-    std::string lib;
-#ifdef _WIN32
-    if (argc > 1) {
-        lib = argv[1];
-    } else {
-        lib = "auto_schedule.dll";
-    }
-
-    if (!LoadLibraryA(lib.c_str())) {
-        DWORD last_err = GetLastError();
-        LPVOID last_err_msg;
-        FormatMessageA(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM |
-                           FORMAT_MESSAGE_IGNORE_INSERTS,
-                       nullptr, last_err, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
-                       reinterpret_cast<LPSTR>(&last_err_msg), 0, nullptr);
-        std::cerr << "Failed to load: " << lib << "\n";
-        std::cerr << "LoadLibraryA failed with error " << last_err << ": "
-                  << static_cast<char *>(last_err_msg) << "\n";
-        LocalFree(last_err_msg);
+    if (!::Halide::Internal::load_plugin("auto_schedule", std::cerr)) {
```

Comment:
Would be nice to have a comment here explaining what "auto_schedule" means.

Reply:
done